packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/User.kt

```diff
@@ -43,14 +43,31 @@ public interface User {
 
     /**
      * The server id of the user.
+     *
+     * This property has been deprecated in favor of [id] and will be replaced in a future release.
      */
+    @Deprecated("Use `User.id` instead", replaceWith = ReplaceWith("id"))
     public val identity: String
 
+    /**
```

Comment:
The reason is that `User.id` is the server ID or primary key (called Identity in the UI) for the entire User. It isn't a `UserIdentity` That user then has a number of `UserIdentity` associated with it, where each is a `Pair(id, provider)`.

The naming is very confusing. I do remember talking about it between all SDK's a few years back, but we didn't have an alternative that was clearly better and since all of this was already established precedent in the docs and in the server UI we decided to keep the naming.

But I agree, not at all confusing 🙈